dataikuapi/dss/ml.py

```diff
@@ -610,7 +713,22 @@ def _pretty_repr(self):
         return self.__class__.__name__ + "(hyperparameter=\"{}\", settings={})".format(self.name, json.dumps(self._algo_settings[self.name], indent=4))
 
     def set_values(self, values=None):
-        self._algo_settings._set_categorical_values(self.name, values=values)
+        if values is None:
+            warnings.warn("Categorical hyperparameter \"{}\" not modified".format(self.name))
```

Comment:
I'm not sure I get why we'd want to support `set_values(None)` that doesn't do anything? Isn't it simpler to just force values to be passed?